ksqldb-engine/src/main/java/io/confluent/ksql/query/PullQueryQueue.java

```diff
@@ -245,6 +259,16 @@ public synchronized boolean acceptRow(final PullQueryRow row) {
     return false;
   }
 
+  private boolean doAcceptRow(final PullQueryRow row) throws InterruptedException {
```

Comment:
specific ask: I'd prefer to do everything in one loop, but if it's better not to, can you either inline this method or extract a method for the other loop?

It looked to me like if we didn't have a limit, we'd only try to insert once, which is not the case.